src/command.c

```diff
@@ -989,3 +1004,121 @@ void dv_run_ipu_command(struct dmp_cmb *cmb, void *bar_logical)
 	}
 	iowrite32(0x1, REG_IO_ADDR(bar_logical, 0x02a0));
 }
+
+// If already a command is converted and not executed yet, -EBUSY is returned.
+static int dv_convert_maximizer_v0(struct device *dev, struct dmp_cmb *cmb,
+			struct dmp_dv_kcmdraw __user *user_cmd, size_t size)
+{
+	struct dmp_cmb_list_entry *cmb_node;
+	struct dmp_dv_kcmdraw_maximizer_v0 cmd;
+	size_t cmd_size = sizeof(uint32_t) * 2 * 8;
+	uint32_t *cmd_buf;
+	uint32_t in_base_addr = 0, in_sz;
+	uint32_t out_base_addr = 0, out_sz;
+	uint32_t num_pixel;
+	uint32_t block_size;
+	uint32_t last_block_src;
+	int i = 0;
+	int ret;
+
+	cmb_node = list_first_entry(&cmb->cmb_list,
+			struct dmp_cmb_list_entry, list_node);
+	if (cmb_node->size != 0)
+		return -EBUSY;
+	if (size < sizeof(struct dmp_dv_kcmdraw_maximizer_v0))
+		return -EINVAL;
+	if (copy_from_user(&cmd, user_cmd, size))
+		return -EFAULT;
+
+	ret = get_dma_addr(dev, cmb, &cmd.input_buf, &in_base_addr, &in_sz);
+	if (ret)
+		return ret;
+	ret = get_dma_addr(dev, cmb, &cmd.output_buf, &out_base_addr, &out_sz);
+	if (ret)
+		return ret;
+
+	// new cmb is the size is smaller
+	if (cmb_node->size + cmd_size + 8 > cmb_node->capacity) {
+		cmb_node = dv_cmb_allocate(dev);
+		if (!cmb_node)
+			return -ENOMEM;
+		list_add(&cmb_node->list_node, &cmb->cmb_list);
+	}
+	cmd_buf = (uint32_t*)((uint8_t*)(cmb_node->logical) + cmb_node->size);
+
+	// fill cmb
+	num_pixel = (uint32_t)cmd.width * (uint32_t)cmd.height;
+	block_size = num_pixel * (cmd.nclass > 8 ? 8 : cmd.nclass) * 2;
+	last_block_src = (in_base_addr + num_pixel * cmd.nclass);
+	if (last_block_src % block_size != 0){
+		last_block_src -= last_block_src % block_size;
+	}
+	cmd_buf[i++] = 0x24;
+	cmd_buf[i++] = ((uint32_t)cmd.nclass << 24) | (num_pixel & 0xffffff);
+	cmd_buf[i++] = 0x28;
+	cmd_buf[i++] = in_base_addr;
+	cmd_buf[i++] = 0x2C;
+	cmd_buf[i++] = out_base_addr;
+	cmd_buf[i++] = 0x30;
+	cmd_buf[i++] = block_size;
+	cmd_buf[i++] = 0x34;
+	cmd_buf[i++] = (num_pixel % 128) * 2 * (cmd.nclass % 8);
+	cmd_buf[i++] = 0x38;
+	cmd_buf[i++] = last_block_src;
+	cmd_buf[i++] = 0;
+	cmd_buf[i++] = 0;
+
+	return 0;
+}
+
+int dv_convert_maximizer_command(struct device *dev, struct dmp_cmb *cmb,
+				struct dmp_dv_kcmd *cmd_info)
```

Comment:
misalignment, either break line just after the bracket or align next line just after the bracket